### PR TITLE
Fixed clippy warnings

### PR DESCRIPTION
**Pull request by devashishdxt**

### CODE REVIEW

File: src/fixed.rs [hunk 1]
```diff
@@ -22,7 +20,7 @@ pub trait FixedInt: Sized + Copy {
         v
     }
     /// Helper: Decode the value from the Vec.
-    fn decode_fixed_vec(v: &Vec<u8>) -> Self {
+    fn decode_fixed_vec(v: &[u8]) -> Self {
```
Comment (dermesser) [inline src/fixed.rs]: please undo this and similar changes, as the Vec is intended (for backwards compatibility at least, I acknowledge the limited usefulness.

If you disagree very strongly, just delete the whole method, as `decode_fixed` already does this.

File: src/fixed.rs [hunk 2]
```diff
@@ -37,23 +35,20 @@ macro_rules! impl_fixedint {
...
-                        Self::REQUIRED_SPACE,
-                    )
+                    std::slice::from_raw_parts(self as *const $t as *const u8, Self::REQUIRED_SPACE)
                 };
             }
 
             fn encode_fixed(self, dst: &mut [u8]) {
                 assert_eq!(dst.len(), Self::REQUIRED_SPACE);
-                let encoded = unsafe { transmute::<&$t, &[u8; $sz]>(&self) };
+                let encoded = unsafe { &*(&self as *const $t as *const [u8; $sz]) };
```
Comment (dermesser) [inline src/fixed.rs]: is this to fix a clippy warning or your personal taste? Because I like the old way better :-)

File: src/fixed.rs (hunk 1 above)
Comment (devashishdxt) [reply src/fixed.rs]: If you need backwards compatibility, I can revert this and mark this function as deprecated. Any thoughts?

File: src/fixed.rs (hunk 2 above)
Comment (devashishdxt) [reply src/fixed.rs]: This is just to fix clippy warning, I don't have a personal taste for this. We can keep it the old way if it suits you better. :-)

**Pull request closed without merging**